### ISSUE TITLE
Enter Sourcegraph search query directly from VS Code command menu

### ISSUE THREAD

**Issue by jjinnii** [sourcegraph/sourcegraph#30054]
#### Feature request description

Ability to enter a search query from VS Code command menu. CMD + Shift + P -> Sourcegraph: search query. 

As a user, if I would like to perform a Sourcegraph search but do not already have a Sourcegraph search tab open, it is an extra step for me to click on the Sourcegraph icon in the toolbar before being able to perform a search. VS Code users are also accustomed to using VS Code commands to quickly perform tasks (especially for extensions), and this may be a gut first step for many of our users. 

#### Additional context

This is taken from[ internal feedback](https://sourcegraph.slack.com/archives/C02E3HHDZ98/p1642441554018800).
 
#### Technical considerations

Are we able to implement auto complete in the VS Code command menu as a user completes a search?

**Comment by tjkandala** [comment]
I initially removed this feature, which exists in the legacy extension, precisely because we can't autocomplete search filters, so I believed that users would rather use the real input box. 

However, this feedback makes a lot of sense. We can't have a fully-featured input box, but we can use the [`QuickPick`](https://code.visualstudio.com/api/references/vscode-api#QuickPick%3CT%3E) API to perhaps show the user's previous searches as well as let them enter a new search query. Would that be good enough?

**Comment by jjinnii** [comment]
Thanks for the context @tjkandala! Just for my understanding, when you say "let them enter a new search query", we wouldn't offer auto-complete, but could a user at least fully type out Sourcegraph syntax? (i.e. repo: wouldn't autocomplete, but a user can type the full syntax out and maybe we could at least color code?)

**Comment by tjkandala** [comment]
@jjinnii I think it's more along the lines of they can type the syntax out but _without_ highlighting, unfortunately. 

However, I just realized that users can just execute the search command in the VS Code command palette which will open or focus the panel... and that's the same number of steps as adding a command that opens a stunted input box.

We also have keyboard shortcuts to open the search panel

**Comment by jjinnii** [comment]
> However, I just realized that users can just execute the search command in the VS Code command palette which will open or focus the panel

Search command meaning ">Sourcegraph: open search tab" right? Totally agree that that resolves most of the user problem I outlined, this is just a nice to have request that I wanted to document, but sounds like it's technically difficult to implement. 

Also to clarify, this is not important/prioritized for first release! I am just documenting feature requests from internal feedback so we have them tracked, but they need to be prioritized. :)

**Comment by abeatrix** [comment]
@jjinnii Looks like this issue is related to https://github.com/sourcegraph/sourcegraph/issues/30267 
Do you think this should be something we can tackle this sprint, or should we wait for more feedback to move forward with both issues?

**Comment by jjinnii** [comment]
Hey @abeatrix! This issue is specific to the VS Code command palette (cmd + shift + p), whereas issue #30267 is much more exploratory to somehow performing a Sourcegraph search in-line to your code blob or maybe even sidebar when you're in file view. 

I think a great start would be if we could have a quick technical prototype for this ability to perform a search query in the command palette and then test it. Based on my earlier messages with @tjkandala, it seems like the effort is not high, but just heavily constrained (no autocomplete, syntax highlighting, etc.), so we should test whether this provides value towards #34346.

Let me know if that makes sense, and whether you could provide some ballpark estimate!

cc: @ryankscott

**Comment by jjinnii** [comment]
Thanks @abeatrix ! Such fast work 🚀 . No, I don't think there's anything to do on #30267 for this iteration, as we need to do both a design and technical exploration on what's possible. 

I think there's a need to clearly communicate the difference between the two ways of searching (direct command palette search or opening the full extension). Copy changes can help a little bit:

1. Sourcegraph: Search with Sourcegraph -> Sourcegraph: Open extension 
2. Sourcegraph: Quick Search -> I _think_ this is smart copy, pulling from other "quick" phrases used in VS Code like Quickpick, etc. This needs to be tested/validated.


The second would be the experience for "quick search". Few questions on this UI:
 
<img width="724" alt="Screen Shot 2022-05-25 at 7 26 37 PM" src="https://user-images.githubusercontent.com/87138876/170242175-b6a0bd45-bfb0-4132-b6d4-e8a7911d4d77.png">

- Is row 1 a standard VS Code pattern for command palette actions that have a sub UI after a user enters an action? 
- Related to above, are the number of rows customizable? For example, could we remove row 1 if we wanted and/or add a row 4? 
- I'm assuming the copy in row 2,3 are customizable?

**Comment by abeatrix** [comment]
As discussed with Sara, more UX works need to be done before we can move forward on the engineering side.

Opened a separate issue to document the prototype created.

**Comment by philipp-spiess** [comment]
@jjinnii Do you know the current state of this issue and if this work is still something we want to tackle?
